File: FrontEnd/lib/domain/model/car.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'car.freezed.dart';
part 'car.g.dart';

@freezed
class Car with _$Car {
  const factory Car({
    @Default('') String idCar,
    @Default('') String idUser,
    @Default('') String idReview,
    @Default('') String nameCar,
    @Default(0) double priceCar,
    @Default('') String fuelTypeCar,
    @Default('') String brandCar,
    @Default('') String colorCar,
    @Default('') String descriptionCar,
    @Default(0) double kilometersCar,
    @Default(0) int seatsCar,
    @Default('') String transmissionCar,
    @Default('') String addressCar,
    @Default(0) double latCar,
    @Default(0) double longCar,
    @Default(0) double averageRating,
    @Default(0) int reviewCount,
    @Default("") String imagesCar,
    @Default('') String statusCar,
    @Default('') String createAt,
  }) = _Car;

  factory Car.fromJson(Map<String, dynamic> json) => _$CarFromJson(json);
}

enum StatusCar {
  renting,
  available,
}

File: BackEnd/models/car.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Car struct {
	IdCar           uuid.UUID `gorm:"primaryKey;type:uuid;default:uuid_generate_v4()"`
	NameCar         string    `gorm:"type:varchar(255);notNull"`
	PriceCar        float64   `gorm:"type:float8;notNull"`
	FuelTypeCar     string    `gorm:"type:string;notNull"`
	BrandCar        string    `gorm:"type:string;notNull"`
	ColorCar        string    `gorm:"type:varchar(255)"`
	DescriptionCar  string    `gorm:"type:text"`
	KilometersCar   float64   `gorm:"type:float8"`
	SeatsCar        int       `gorm:"type:int"`
	TransmissionCar string    `gorm:"type:text"`
	AddressCar      string    `gorm:"type:string;notNull"`
	LatCar          float64   `gorm:"type:float8;notNull"`
	LongCar         float64   `gorm:"type:float8;notNull"`
	ImagesCar       string    `gorm:"type:text"`
	StatusCar       string    `gorm:"type:string"`
	CreatedAt       time.Time `gorm:"notNull;default:CURRENT_TIMESTAMP"`
	UpdatedAt       time.Time `gorm:"notNull;default:CURRENT_TIMESTAMP"`
	UserId          uuid.UUID `gorm:"type:uuid;notNull;foreignKey:UserId"`
	ReviewId        uuid.UUID `gorm:"type:uuid;"`

	User        *User
	CarReviews  []CarReview
	CarRentails []CarRentail
}
